src/Microsoft.TestPlatform.PlatformAbstractions/common/System/ProcessHelper.cs

```diff
@@ -127,10 +127,9 @@ public void SetExitCallback(int processId, Action callbackAction)
         }
 
         /// <inheritdoc/>
-        public void TerminateProcess(int processId)
+        public void TerminateProcess(object process)
         {
-            var process = Process.GetProcessById(processId);
-            process.Kill();
+            ((Process)process).Kill();
```

Comment:
please add a null check after Typecasting, if something else is passed other than process object